### ISSUE TITLE
WebFinger: remoteStorage 'properties' need to be absolute URIs (and spec update)

### ISSUE THREAD

**Issue by silverbucket** [remotestorage/remotestorage.js#271]
I got a heads up email from Paul E. Jones, who's one of the main guys working on the WebFinger spec. 

tl;dr - we need to set up URI endpoints for our remoteStorage properties, and update the remoteStorage WebFinger query to look for absolute URIs.

Here's the full email from him:

---

I was reviewing some of my documentation for WebFinger and I noted an error that had persisted in many drafts and my own documentation.  Specifically, I had been using simple tokens to refer to property names within the context of a link.

I queried your server and you are doing the same:

``` javascript
    {
      "href": "https://heahdk.net/storage/silverbucket",
      "rel": "remotestorage",
      "type": "https://www.w3.org/community/rww/wiki/read-write-web-00#simple",
      "properties": {
        "auth-method": "http://tools.ietf.org/html/rfc6749#section-4.2",
        "auth-endpoint": "https://heahdk.net/authorizations/new?login=silverbucket"
      }
    }
```

I actually corrected mine to align with what the WebFinger spec said, which is derived from the XRD spec.  Specifically, type property names are supposed to be absolute URIs, like this:

``` javascript
    {
      "rel" : " http://packetizer.com/rel/smtp-server",
      "properties" :
      {
        "http://packetizer.com/ns/host" : "dublin.packetizer.com",
        "http://packetizer.com/ns/port" : "587",
        "http://packetizer.com/ns/login-required" : "yes",
        "http://packetizer.com/ns/transport" : "starttls"
      }
  }
```

I actually do have a preference for using simple token values, since it is far less verbose.  On the other hand, changing that might risk interop with any existing RFC 6415 entity.  Here’s a pointer to the XRD spec:
http://docs.oasis-open.org/xri/xrd/v1.0/xrd-1.0.html#element.property

It says, “This value MUST be an absolute URI.” “Value”, here, refers to the name of the property.  The XML data type is actually “anyURI”, but the prose restricts it to being an absolute URI.

One of the things I believe we need to do next is to set about standardizing these link relations used in WebFinger and associated properties.  I’ve been defining link relation values under packetizer.com, since I don’t control the webfinger.net domain and have no other appropriate place to define them right now.  I think using standard values would be better and property names could be of the form urn:ietf:webfinger:port, for example.

I’m not sure what’s best here, but I’d really like to see Cyrus take up the initiative he’s doing to define mail server configuration in that way.

---

If there is a spec for this rel type (and the "rel" can be either a token registered with IANA or an absolute URI), then that spec should define all of the properties.  What absolute URI form to take would be up to those writing the spec.  It could be http:, urn:, or something else.
